components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -345,16 +359,90 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                         app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
                         app.setTokenRevocationWithIDPSessionTerminationEnabled(
                                 application.isTokenRevocationWithIDPSessionTerminationEnabled());
-                        app.setTokenEndpointAuthMethod(application.getTokenEndpointAuthMethod());
-                        app.setTokenEndpointAuthSignatureAlgorithm(
-                                application.getTokenEndpointAuthSignatureAlgorithm());
-                        app.setSectorIdentifierURI(application.getSectorIdentifierURI());
-                        app.setIdTokenSignatureAlgorithm(application.getIdTokenSignatureAlgorithm());
-                        app.setRequestObjectSignatureAlgorithm(application.getRequestObjectSignatureAlgorithm());
+                        String tokenEndpointAuthMethod = application.getTokenEndpointAuthMethod();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+                            if (enforceFAPIDCR) {
+                                validateFAPITokenAuthMethods(tokenEndpointAuthMethod);
+                            } else {
+                                filterTokenEndpointAuthMethods(tokenEndpointAuthMethod);
+                            }
+                            app.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
+                        }
+                        String tokenEndpointAuthSigningAlgorithm = application.getTokenEndpointAuthSignatureAlgorithm();
+                        if (StringUtils.isNotEmpty(tokenEndpointAuthSigningAlgorithm)) {
+                            if (enforceFAPIDCR) {
+                                validateFAPISignatureAlgorithms(tokenEndpointAuthSigningAlgorithm);
+                            } else {
+                                filterSignatureAlgorithms(tokenEndpointAuthSigningAlgorithm,
+                                      OAuthConstants.TOKEN_EP_SIGNATURE_ALG_CONFIGURATION);
+                            }
+                            app.setTokenEndpointAuthSignatureAlgorithm(tokenEndpointAuthSigningAlgorithm);
+                        }
+                        if (StringUtils.isNotEmpty(application.getSubjectType())) {
```

Comment:
Is the value is empty shouldn't we set the default?

Reply:
If the application doesn't have this value its the default value that is sent in the id token... but will change to default here as well....